enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
What exactly is a duplicated init-container ? An init-container with the same name or with the same content ? Completely agree that we should ignore a second init-container if it has the same content (with a warning), but would always croak if they have the same name but different content. If not, that will **break** things eventually even with valid Kubernetes resource (because of a missing init-container then).

Let me add that check for equality of the content, so that we are on the safe side. This would cover your use case, too (but [as said](https://github.com/fabric8io/fabric8-platform/pull/142#issuecomment-277751283) there is a simple solution without code change.